examples/diagnostics/TestWFS.ipynb

```diff
@@ -0,0 +1,359 @@
+    "    #now create an SRW optical container with a drift of length pd\n",
+    "    optDrift=srwlib.SRWLOptD(sd)\n",
+    "    propagParDrift = [0, 0, 1., 0, 0, 1, 1, 1, 1, 0, 0, 0]\n",
+    "    optBLW = srwlib.SRWLOptC([optDrift],[propagParDrift])\n",
+    "    srwlib.srwl.PropagElecField(thisSlice.wfr, optBLW)\n",
+    "        \n",
+    "    wflist.append(thisSlice.wfr)\n",
+    "    \n",
+    "#Now add wavefronts together\n",
+    "wfr = copy.deepcopy(wflist[0])\n",
+    "for slice_index in np.arange(nslice-1):\n",
+    "    wfr.addE(wflist[slice_index + 1])\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2980f3ef-0a79-4cd5-8a6a-b3b78a61cc3b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Plot the intensity of the resulting added wavefront at the wavefront sensor\n",
+    "\n",
+    "arIinit = srwlib.array('f', [0]*wfr.mesh.nx*wfr.mesh.ny) # \"flat\" array to take 2D intensity data\n",
+    "srwl.CalcIntFromElecField(arIinit, wfr, 0, 0, 3, wfr.mesh.eStart, 0, 0) #extracts intensity\n",
+    "\n",
+    "##Reshaping electric field data from flat to 2D array\n",
+    "arIinit2D = np.array(arIinit).reshape((wfr.mesh.nx, wfr.mesh.ny), order='C')\n",
+    "wfrsizei=np.size(arIinit)\n",
+    "\n",
+    "print('Size of initial wavefront data array (coordinate):',np.shape(arIinit2D))\n",
+    "xvalsf02=np.linspace(wfr.mesh.xStart,wfr.mesh.xFin,wfr.mesh.nx)\n",
+    "yvalsf02=np.linspace(wfr.mesh.yStart,wfr.mesh.yFin,wfr.mesh.ny)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "15be12e2-89b8-4b53-b87c-8c94b171956e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# plot the computed intensity\n",
+    "\n",
+    "with plt.style.context(('seaborn-poster')):\n",
+    "    fig = plt.figure(figsize=(12,7))\n",
+    "    ax = fig.gca()\n",
+    "\n",
+    "    # By setting wfr0a.unitElFldAng=1, default units should  now be in mrad(?)\n",
+    "    plt.pcolormesh(np.multiply(xvalsf02,1e6), np.multiply(yvalsf02,1e6), arIinit2D, cmap=plt.cm.viridis,shading='auto')\n",
+    "    plt.colorbar()\n",
+    "    #['Horizontal Position [mm]', , 'Intensity']\n",
+    "    ax.set_ylabel(r'Vertical Position [$\\mu m$]')\n",
+    "    ax.set_xlabel(r'Horizontal Position [$\\mu m$]')\n",
+    "    ax.set_title('Intensity before the crystal')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ffa05073-4992-4653-9de6-51858c86d74e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# plot the phase\n",
+    "\n",
+    "with plt.style.context(('seaborn-poster')):\n",
+    "    fig = plt.figure(figsize=(12,7))\n",
+    "    ax = fig.gca()\n",
+    "\n",
+    "    # By setting wfr0a.unitElFldAng=1, default units should  now be in mrad(?)\n",
+    "    plt.pcolormesh(np.multiply(xvalsf02,1e6), np.multiply(yvalsf02,1e6), arPhase2D, cmap=plt.cm.viridis,shading='auto')\n",
```

Comment:
```arPhase2D``` is not defined before used here, so this line runs an error. 
But otherwise looks good.

Reply:
Thanks for catching that @gurhar1133 ! Not sure how it ended up in that state. I fixed it and pushed to the branch.

Reply:
np I think it can be easy to work on a cells then going back and forth in a way that can mess with the program flow from cell to cell